### PR TITLE
statcast large-query timing script in CI

### PR DESCRIPTION
**Pull request by bdilday**
This PR adds a step to the CI that will execute a large query against statcast and check that it completes within a given time threshold. This will help make sure subsequent PRs are taking care of https://github.com/jldbc/pybaseball/issues/154

### CODE REVIEW

File: scripts/statcast_timing.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    if len(sys.argv) == 2:
+        time_threshold = float(sys.argv[1])
+    else:
+        time_threshold = EXPECTED_TIME_THRESHOLD
+
+    start_time = time.time()
+    _ = statcast(START_DATE, END_DATE)
+    end_time = time.time()
+    query_time = end_time-start_time
+    if query_time > time_threshold:
```
Comment (schorrm) [inline scripts/statcast_timing.py:19]: Do you know if subprocess works in CI? We could run this in a subprocess with a timeout and kill if it went too far over (let's say over a minute).

**Comment by schorrm** [comment]
LGTM though, just waiting to hear your thoughts on capping the runtime for this

File: scripts/statcast_timing.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        sys.stderr.write("query took {:.2f} seconds "
```
Comment (TheCleric) [inline scripts/statcast_timing.py:20]: Should we use f-strings instead of a format call?

File: scripts/statcast_timing.py (hunk 1 above)
Comment (bdilday) [reply scripts/statcast_timing.py:19]: no I don't know. the goal is to make CI run faster? it feels like solving a problem we don't have.

File: scripts/statcast_timing.py (hunk 2 above)
Comment (bdilday) [reply scripts/statcast_timing.py:20]: yes good idea - I wasn't aware (until I just looked it up now) that you can use a format-code with f-strings.

Comment (bdilday) [reply scripts/statcast_timing.py]: I used part f-strings, part `format`, let me know if loosk o0k

Comment (TheCleric) [reply scripts/statcast_timing.py]: It looks okay. You could also do this:

```python
sys.stdout.write(f"query took {query_time: .1f} seconds which is {"less" if query_time < time_threshold else "greater"} than threshold of {time_threshold: .1f}\n")
```

Comment (schorrm) [reply scripts/statcast_timing.py]: For that '{"less" if query_time < time_threshold else "greater"}'
Should be move outside. No long stuff in there

File: scripts/statcast_timing.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+
+def main():
+    args = _parse_args()
+    start_time = time.time()
+    _ = statcast(args.start_date, args.end_date)
+    end_time = time.time()
+    query_time = end_time-start_time
+    threshold_exceeded = query_time > args.time_threshold
+    print(f"query took {query_time: .1f} (expected less than {args.time_threshold: .1f})")
```
Comment (TheCleric) [inline scripts/statcast_timing.py:26]: Even better! 😄

File: scripts/statcast_timing.py (hunk 1 above)
Comment (bdilday) [reply scripts/statcast_timing.py]: @schorrm I looked up the [documentation on this](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes) and implemented a 1 minute timeout on the large-query step

**Pull request merged**